roaring/roaring.go

```diff
@@ -503,18 +520,60 @@ func (b *Bitmap) countEmptyContainers() int {
 	return result
 }
 
+// Optimize converts array and bitmap containers to run containers as necessary.
+func (b *Bitmap) Optimize() {
+	for _, c := range b.containers {
+		c.Optimize()
+	}
+}
+
 // WriteTo writes b to w.
 func (b *Bitmap) WriteTo(w io.Writer) (n int64, err error) {
+	b.Optimize()
 	// Remove empty containers before persisting.
 	//b.removeEmptyContainers()
 	containerCount := len(b.keys) - b.countEmptyContainers()
 
+	// Create bitset indicating runs, record whether any runs present.
+	containsRuns := false
+	runFlagBitset := make([]uint8, (containerCount+7)/8)
+	k := 0
+	for _, c := range b.containers {
+		if c.n == 0 {
```

Comment:
It's no longer obvious to me what I meant when I suggested testing this, and `TestBitmap_BitmapWriteToWithEmpty` already tests writing+reading a bitmap with an empty container. @tgruben @jaffee is there something else worth testing here?